File: client/jetbrains/src/main/java/com/sourcegraph/cody/context/RepoAvailableEmbeddingStatus.java
package com.sourcegraph.cody.context;

import org.jetbrains.annotations.NotNull;

public abstract class RepoAvailableEmbeddingStatus implements EmbeddingStatus {
  private final String simpleRepositoryName;

  protected RepoAvailableEmbeddingStatus(String fullRepositoryName) {
    this.simpleRepositoryName = getRepositoryNameAfterLastSlash(fullRepositoryName);
  }

  @NotNull
  private static String getRepositoryNameAfterLastSlash(String fullRepositoryName) {
    int indexOfLastSlash = fullRepositoryName.lastIndexOf('/');
    String repoNameWithoutTrailingSlash =
        indexOfLastSlash == fullRepositoryName.length() - 1
            ? fullRepositoryName.substring(0, indexOfLastSlash)
            : fullRepositoryName;
    indexOfLastSlash = repoNameWithoutTrailingSlash.lastIndexOf('/');
    return indexOfLastSlash != -1 && indexOfLastSlash != repoNameWithoutTrailingSlash.length() - 1
        ? repoNameWithoutTrailingSlash.substring(indexOfLastSlash + 1)
        : repoNameWithoutTrailingSlash;
  }

  @Override
  public @NotNull String getMainText() {
    return simpleRepositoryName;
  }
}

File: client/jetbrains/src/main/java/com/sourcegraph/cody/context/RepoAvailableEmbeddingStatus.kt
package com.sourcegraph.cody.context

abstract class RepoAvailableEmbeddingStatus protected constructor(fullRepositoryName: String) :
    EmbeddingStatus {
  private val simpleRepositoryName: String

  init {
    simpleRepositoryName = getRepositoryNameAfterLastSlash(fullRepositoryName)
  }

  override fun getMainText(): String {
    return simpleRepositoryName
  }

  companion object {
    private fun getRepositoryNameAfterLastSlash(fullRepositoryName: String): String {
      var indexOfLastSlash = fullRepositoryName.lastIndexOf('/')
      val repoNameWithoutTrailingSlash =
          if (indexOfLastSlash == fullRepositoryName.length - 1)
              fullRepositoryName.substring(0, indexOfLastSlash)
          else fullRepositoryName
      indexOfLastSlash = repoNameWithoutTrailingSlash.lastIndexOf('/')
      return if (indexOfLastSlash != -1 &&
          indexOfLastSlash != repoNameWithoutTrailingSlash.length - 1)
          repoNameWithoutTrailingSlash.substring(indexOfLastSlash + 1)
      else repoNameWithoutTrailingSlash
    }
  }
}
